src/shared/resolvers/types.ts

```diff
@@ -45,6 +50,52 @@ export const collectionPartnershipFieldResolvers = {
   },
 };
 
+/**
+ * Field-level resolvers for the Label type.
+ *
+ * Just like with the CollectionPartnership resolvers above, querying two fields
+ * on the label entity doesn't result in two separate queries.
+ *
+ * In both cases, for a query that returns labels directly from the database
+ * (e.g. `labels`), use the existing value from the Label table.
+ *
+ * For a query that goes through the CollectionLabel relationship table first,
+ * look up the values for the corresponding label in its own table.
+ */
+export const collectionLabelsFieldResolvers = {
+  async externalId(
+    parent: Label | CollectionLabel,
+    _,
+    { db }
+  ): Promise<string> {
+    // TypeScript is unhappy while running integration tests without these checks in place
+    if ('externalId' in parent) {
+      return parent.externalId;
+    }
+
+    if ('labelId' in parent) {
+      const label = await getLabelById(db, parent.labelId);
+      return label.externalId;
+    }
```

Comment:
i wouldn't be mad if at some point someone wanted to add comments on these `if` statements making it obvious which satisfies which query condition (querying a label directly vs getting labels through a collection query).

Reply:
@jpetto, added more comments in https://github.com/Pocket/collection-api/pull/1003! :)